src/operator/tensor/elemwise_unary_op.h

```diff
@@ -476,6 +476,31 @@ void HardSigmoidBackward(const nnvm::NodeAttrs& attrs,
   });
 }
 
+struct ReshapeLikeParam : public dmlc::Parameter<ReshapeLikeParam> {
+  int lhs_begin, rhs_begin;
```

Comment:
I wonder if these two parameters should be made optional too, given that reshape_like op has been around for 10 months.

Reply:
They are optional in the sense that they have default values `DMLC_DECLARE_FIELD(lhs_begin).set_default(0)` that match the old behaviour if not explicitly specified. I thought `dmlc::optional<int>` was simply for the case where you had to handle the `None`-value case, which is not necessary to support for `lhs_begin` and `rhs_begin`. Note that the same is done for `slice_axis`, `begin` is an `int` and `end` is `dmlc::optional<int>`.

Or am I misunderstanding something?

Reply:
Considering the case where the parameters may be in some serialized format, it may be necessary to support null values to ensure compatibility there too.

Reply:
@szha Can I clarify something? Maybe I am misunderstanding. Are you saying that it is required to make new parameters on existing layers optional<...> for backward compatibility reasons, *even* if they have defaults?

Reply:
@taliesinb I think it doesn't hurt in this case and was suggesting that we lean on the safer side. There might be cases where the default value filling on the frontend fails, such as deserializing a graph.

Reply:
@szha ok. so it sounds like there isn't a particular policy at the moment about how to add new parameters to existing layers. do you mind if i ask on the mailing list about what such a policy should be, just for the next time this happens?

Reply:
of course not. good idea :)

Reply:
@szha: I've changed `lhs_begin` and `rhs_begin` to use optional (and added tests for this case). Can we merge?